rsocket-examples/src/main/java/io/rsocket/examples/transport/tcp/plugins/LimitRateInterceptoreExample.java

```diff
@@ -0,0 +1,135 @@
+package io.rsocket.examples.transport.tcp.plugins;
+
+import io.rsocket.Payload;
+import io.rsocket.RSocket;
+import io.rsocket.core.RSocketConnector;
+import io.rsocket.core.RSocketServer;
+import io.rsocket.examples.transport.tcp.stream.StreamingClient;
+import io.rsocket.plugins.RSocketInterceptor;
+import io.rsocket.transport.netty.client.TcpClientTransport;
+import io.rsocket.transport.netty.server.TcpServerTransport;
+import io.rsocket.util.DefaultPayload;
+import io.rsocket.util.RSocketProxy;
+import java.time.Duration;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.concurrent.Queues;
+
+public class LimitRateInterceptoreExample {
+
+  private static final Logger logger = LoggerFactory.getLogger(StreamingClient.class);
+
+  public static void main(String[] args) {
+    BlockingQueue<String> requests = new ArrayBlockingQueue<>(100);
+    RSocketServer.create(
+            (sp, sr) ->
+                Mono.just(
+                    new RSocket() {
+                      @Override
+                      public Flux<Payload> requestStream(Payload payload) {
+                        return Flux.interval(Duration.ofMillis(100))
+                            .doOnRequest(e -> requests.add("Responder requestN(" + e + ")"))
+                            .map(aLong -> DefaultPayload.create("Interval: " + aLong));
+                      }
+
+                      @Override
+                      public Flux<Payload> requestChannel(Publisher<Payload> payloads) {
+                        return Flux.from(payloads)
+                            .doOnRequest(e -> requests.add("Responder requestN(" + e + ")"));
+                      }
+                    }))
```

Comment:
This could be `SocketAcceptors.with(new RSocket() {...})`.

Reply:
fixed